crates/era-downloader/src/stream.rs

```diff
@@ -0,0 +1,216 @@
+use crate::EraClient;
+use futures_util::{stream::FuturesOrdered, FutureExt, Stream, StreamExt};
+use pin_project::pin_project;
+use reqwest::Url;
+use std::{
+    collections::VecDeque,
+    fmt::{Debug, Formatter},
+    future::Future,
+    path::Path,
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+/// An asynchronous stream of ERA1 files.
+#[derive(Debug)]
+pub struct EraStream {
+    download_stream: DownloadStream,
+    starting_stream: StartingStream,
+}
+
+impl EraStream {
+    /// Constructs a new [`EraStream`] that downloads concurrently up to `max_concurrent_downloads`
+    /// ERA1 files to `client` `folder`, keeping their count up to `max_files`.
+    pub fn new(client: EraClient, max_files: usize, max_concurrent_downloads: usize) -> Self {
```

Comment:
can we convert the two args into a `struct EraStreamConfig` type, in case we will need more settings?